### PR TITLE
Swap SGD for Adam in benchmarks

### PR DESCRIPTION
**Pull request by mlazos**
Swap Adam in for the optimizer in our benchmarks

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -979,7 +979,7 @@ def setup_amp(self):
 
     def init_optimizer(self, name, device, params):
         if device == "cuda" and self.args.training and name not in CI_SKIP_OPTIMIZER:
-            self.optimizer = torch.optim.SGD(params, lr=0.01)
```
Comment (vadimkantorov) [inline benchmarks/dynamo/common.py:982]: interesting, SGD and Adam have different treatment of defaults wrt learning rate?

Comment (mlazos) [reply benchmarks/dynamo/common.py:982]: Yeah, SGD is required, Adam has a default, not sure why lol
